### PR TITLE
Laundry list to re-align after PEP updates

### LINKED ISSUE #153: Default chron schedule for GHA
The cookiecutter current sets up to run GHA daily by default. We should reduce this - maybe to weekly by default.

### PR DESCRIPTION
**Pull request by Lnaden**
Follow up to #152

* Update README to include information about `pyproject.toml` and add @eirrgang as a major contributor
* Removed references to Python 2.7 in the readme, its been long enough.
* Update the reference CI's to reflect the minor python version bump
* Update pseudo-validate to reflect minor python version bumps
* Using checkout v3 action now
* GHA Cron runs weekly now (Fixes #153)
* The `git init` which runs sets `--initial-branch=main`, no more `master`
* CI both CC-level and generated level only reference main now, no more master
* Removed the `legacy-miniconda-setup` folders, it was time for me to let that go
* Bump CC version to 1.9

### CODE REVIEW

**Comment by Lnaden** [comment]
@eirrgang @janash This all look good to you?

**Comment by eirrgang** [comment]
Doh! I just made comments on the commit instead of the PR... https://github.com/Lnaden/cookiecutter-cms/commit/4e5bf1a8568aa5a0682267d1f0a76feb8b841965

**Comment by eirrgang** [comment]
> Lnaden@4e5bf1a

tldr:
* Do we want to comment on `_version.py` as a _generated_ file?
* This obviates the workshop section on changing the default branch name from `master` to `main`. Should we punt the updates to Sam?

**Review by eirrgang — APPROVED** [review]

**Comment by eirrgang** [comment]
Oh, one more thing: Don't the docs in https://github.com/MolSSI/cookiecutter-cms/blob/main/docs/index.rst warrant corresponding updates?

**Comment by Lnaden** [comment]
> Do we want to comment on _version.py as a generated file?

Yes

> This obviates the workshop section on changing the default branch name from master to main. Should we punt the updates to Sam?

I'll make a PR there in response to this

> Don't the docs in 

Yes. I'll handle it.

**Review by eirrgang — APPROVED** [review]
lgtm!
